src/adapter/etl-adapter-json/src/Flow/ETL/Adapter/JSON/JSONMachine/JsonExtractor.php

```diff
@@ -91,6 +102,30 @@ public function withSchema(Schema $schema) : self
         return $this;
     }
 
+    private function getIterator()
```

Comment:
I would put a return type here `\Generator`

Reply:
Thats quick, i had that initially but get this type error

```
TypeError: Flow\ETL\Adapter\JSON\JSONMachine\JsonExtractor::getIterator(): Return value must be of type Generator, Closure returned
```
I can dig a bit deeper

Reply:
I've changed it to return "callable" which i guess is correct as its returning a function that will generate an iterator. Maybe getIterator is not the best name

Reply:
I think that if we split JsonLines and regular Json extractors/loaders this problem might even go away

Reply:
Yes I did think that might be worthwhile as well, having a seperate JsonLExtractor and then using ```from_jsonl``` in the DSL.

```
                schema: $schema = df()
                    ->read(from_jsonl(__DIR__ . '/../../Fixtures/timezones.jsonl')))
```

I think JSONL still needs two iterators one for when its pointing at an Array Sub-entry and the case where there is no pointer and its reading as an Object; the iterator from JSONMachine behaves very differently in these cases.

Reply:
> I think JSONL still needs two iterators one for when its pointing at an Array Sub-entry and the case where there is no pointer and its reading as an Object; the iterator from JSONMachine behaves very differently in these cases.

hmm would it be possible to just execute that callable and return the result which would be a generator (iterable)?